### PR TITLE
refactor: input amount

### PR DESCRIPTION
**Pull request by helciofranco**
`InputAmount` is now following the latest pattern.

Screenshot 2024-02-23 at 16 25 38

### Usage
```tsx
<InputAmount
  disabled={disabled}
  balance={balance}
  value={amount}
  onChange={(bn) => console.log(bn)}
>
  <Flex>
    <InputAmount.Field />
    <InputAmount.Slot>
      <InputAmount.MaxBalance />
      <InputAmount.CoinSelector
        asset={asset}
        onClick={openAssetsDialog}
      />
    </InputAmount.Slot>
  </Flex>
  <InputAmount.Balance />
</InputAmount>
```

### CODE REVIEW

File: packages/ui/src/components/InputAmount/InputAmount.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+const meta: Meta<typeof InputAmount> = {
+  title: 'Form/InputAmount',
+  component: InputAmount,
+};
+
+export default meta;
+type Story = StoryObj<typeof InputAmount>;
+
+export const Usage: Story = {
```
Comment (pedronauck) [inline packages/ui/src/components/InputAmount/InputAmount.stories.tsx:14]: would be nice to have couple more cases for the story, change variations to show that is able to show/hide child components

File: packages/ui/src/components/InputAmount/InputAmount.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      balance={undefined}
+      value={undefined}
+      disabled={false}
+      onChange={(balance) => console.log(balance)}
+    >
+      <Flex>
+        <InputAmount.Field placeholder="0.00" />
+
+        <InputAmount.Slot>
+          <InputAmount.MaxBalance
```
Comment (pedronauck) [inline packages/ui/src/components/InputAmount/InputAmount.stories.tsx:28]: I don't know, maybe we could use `InputAmount.ButtonMaxBalance` or `InputAmount.SetMaxBalance`, what do you think?!

**Review by pedronauck — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/InputAmount/InputAmount.stories.tsx (hunk 2 above)
Comment (helciofranco) [reply packages/ui/src/components/InputAmount/InputAmount.stories.tsx]: https://github.com/FuelLabs/fuel-explorer/pull/192/commits/197e081a458382b61bd5a416c678dcfe44c01b8f

File: packages/ui/src/components/InputAmount/InputAmount.stories.tsx (hunk 1 above)
Comment (helciofranco) [reply packages/ui/src/components/InputAmount/InputAmount.stories.tsx:14]: https://github.com/FuelLabs/fuel-explorer/pull/192/commits/09164e159cf14ab1ee0975a7ef59d4eba60d2415

**Review by pedronauck — APPROVED** [review]

**Pull request merged**